docker/reference/reference.go

```diff
@@ -102,13 +107,13 @@ func WithTag(name Named, tag string) (NamedTagged, error) {
 
 // WithDigest combines the name from "name" and the digest from "digest" to form
 // a reference incorporating both the name and the digest.
-func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
-	r, err := distreference.WithDigest(name, digest)
-	if err != nil {
-		return nil, err
-	}
-	return &canonicalRef{namedRef{r}}, nil
-}
+//func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
```

Comment:
(No, this can happen even when `vendor` is not in `$GOPATH`. the `/vendor/` is added into the type names by the `GO15VENDOREXPERIMENT` automatic vendoring as described in the Google doc linked above.

Basically, that vendoring design only works when vendoring happens at the “highest-level user”; it is not possible for two independent components to vendor the same dependency, and then to import those two independent components into a single project and expect the two vendored instances of the dependency to cooperate.

docker/distribution is intended as a standalone highest-level server, so it is doing vendoring for itself, while containers/image (and docker/docker) is using docker/distribution as a library. docker/docker resolves this by modifying the vendored docker/distribution copy to drop docker/distribution/vendor, we resolve it by the above workaround.)